include/envoy/server/filter_config.h

```diff
@@ -105,6 +105,8 @@ class ServerFactoryContext : public virtual CommonFactoryContext {
   ~ServerFactoryContext() override = default;
 };
 
+class TransportSocketFactoryContext;
```

Comment:
will include cause cyclic dependency?

Reply:
Had to test it, no it will not. However, an include is not needed here, and checking out I see that other places in `include` do the same forward declaration (e.g., `include/envoy/secret/secret_manager.h`).

Reply:
Hmm, if it won't then let's `#include` per [style guide](https://google.github.io/styleguide/cppguide.html#Forward_Declarations).

Reply:
@lizan Done.